scylla-cdc-replicator/src/replication_tests.rs

```diff
@@ -562,4 +664,70 @@ mod tests {
             .await
             .unwrap();
     }
+
+    #[tokio::test]
+    #[should_panic]
+    async fn test_checking_timestamps() {
+        let schema = TestTableSchema {
+            name: "COMPARE_TIME".to_string(),
+            partition_key: vec![("pk", "int")],
+            clustering_key: vec![("ck", "int")],
+            other_columns: vec![("v1", "int"), ("v2", "boolean")],
+        };
+
+        let operations = vec![
+            (
+                "INSERT INTO COMPARE_TIME (pk, ck, v1, v2) VALUES (?, ?, ?, ?)",
+                vec![Int(1), Int(2), Int(3), Boolean(true)],
+            ),
+            (
+                "UPDATE COMPARE_TIME SET v2 = ? WHERE pk = ? AND ck = ?",
+                vec![Boolean(false), Int(1), Int(2)],
+            ),
+        ];
+
+        let session = Arc::new(
+            SessionBuilder::new()
+                .known_node(&get_uri())
+                .build()
+                .await
+                .unwrap(),
+        );
+        setup_tables(&session, &schema).await.unwrap();
+        execute_queries(&session, operations).await.unwrap();
+
+        replicate(&session, &schema.name).await.unwrap();
+
+        compare_changes(&session, &schema.name).await.unwrap();
+        // We update timestamps for v2 column in src.
+        session
+            .query(
+                "UPDATE COMPARE_TIME SET v2 = ? WHERE pk = ? AND ck = ?",
+                (false, 1, 2),
+            )
+            .await
+            .unwrap();
+        compare_timestamps(&session, &schema).await.unwrap();
+    }
+
+    #[tokio::test]
+    async fn test_skipping_non_readable_timestamps() {
```

Comment:
The test also contains types which do have readable timestamps, e.g. `frozen<set<frozen<set<frozen<set<int>>>>>>` and `tuple<map<int, int>>`. Note that if you nest a collection type in a `tuple`, the entire thing is frozen because `tuple`s cannot be non-frozen.

I like that the test contains all kinds of types, but the test name is confusing - some columns are readable.

Reply:
I have changed this to `test_whether_writetime_function_fails_for_complicated_types`. I hope it's better now.